src/ui/components/SecondaryToolbox/TerminalContextAdapter.tsx

```diff
@@ -0,0 +1,50 @@
+import {
+  NewTerminalExpression,
+  TerminalContext,
+  TerminalExpression,
+} from "bvaughn-architecture-demo/src/contexts/TerminalContext";
+import React, {
+  PropsWithChildren,
+  useCallback,
+  useMemo,
+  useRef,
+  useState,
+  useTransition,
+} from "react";
+
+export default function TerminalContextController({ children }: PropsWithChildren) {
+  const [isPending, startTransition] = useTransition();
+  const [messages, setMessages] = useState<TerminalExpression[]>([]);
+
+  const idCounterRef = useRef(0);
+
+  const addMessage = useCallback((partialTerminalExpression: NewTerminalExpression) => {
+    // New expressions should be added in a transition because they suspend.
+    startTransition(() => {
+      startTransition(() => {
```

Comment:
Do we need two nested `startTransition()` calls here? If so, why?

Reply:
No, def not. This looks like a rebase flub. Thanks!